### PR TITLE
Added System.GC.GetTotalAllocatedBytes to the ref assembly

### PR DESCRIPTION
**Pull request by VSadov**
CoreFx part for https://github.com/dotnet/coreclr/pull/23852

### CODE REVIEW

File: src/System.Runtime/src/ApiCompatBaseline.uapaot.txt [hunk 1]
```diff
@@ -18,3 +18,5 @@ CannotRemoveBaseTypeOrInterface : Type 'System.Memory<T>' does not implement int
 CannotRemoveBaseTypeOrInterface : Type 'System.ReadOnlyMemory<T>' does not implement interface 'System.IEquatable<System.ReadOnlyMemory<T>>' in the implementation but it does in the contract.
 MembersMustExist : Member 'System.GC.GetGCMemoryInfo()' does not exist in the implementation but it does exist in the contract.
 TypesMustExist : Type 'System.GCMemoryInfo' does not exist in the implementation but it does exist in the contract.
+MembersMustExist : Member 'System.GC.GetTotalAllocatedBytes(System.Boolean)' does not exist in the implementation but it does exist in the contract.
+
```
Comment (ahsonkhan) [inline src/System.Runtime/src/ApiCompatBaseline.uapaot.txt]: nit: remove new line

File: src/System.Runtime/ref/System.Runtime.cs [hunk 2]
```diff
@@ -1273,6 +1273,7 @@ public static partial class GC
         public static GCMemoryInfo GetGCMemoryInfo() { throw null; }
         public static int GetGeneration(object obj) { throw null; }
         public static int GetGeneration(System.WeakReference wo) { throw null; }
+        public static long GetTotalAllocatedBytes(bool precise = false) { throw null; }
```
Comment (stephentoub) [inline src/System.Runtime/ref/System.Runtime.cs]: Can you add some basic tests for this?

Comment (VSadov) [reply src/System.Runtime/ref/System.Runtime.cs]: Yes. Will add. 
We have tests on the coreclr side, but having something trivial in CoreFX would be useful too.

**Comment by VSadov** [comment]
The first iteration failed to build. I assume the bits from coreclr were not yet in corefx.

**Comment by VSadov** [comment]
hmm. It is passing too fast. Is it running any tests?

**Comment by VSadov** [comment]
now it looks like it is running something (after close/open)

**Comment by ViktorHofer** [comment]
Will this solve https://github.com/dotnet/corefx/pull/37670?

**Comment by VSadov** [comment]
@ViktorHofer - probably. Although I expected it to be the other way - I thought it is this PR that is blocked until #37670 is merged.

**Comment by VSadov** [comment]
It looks like a mismatch in either way results in a failure, so one of this PRs will need to be merged with failures, or somehow both merged as a single PR.

**Comment by VSadov** [comment]
Maybe if I rebase my change onto https://github.com/dotnet/corefx/tree/darc-master-2abf1dcb-4b10-4c99-adcc-555d00747083 everything will work?

**Comment by VSadov** [comment]
I have rebased this change onto `darc-master-2abf1dcb-4b10-4c99-adcc-555d00747083` , so it now has both #37670 and the ref changes.

**Comment by VSadov** [comment]
Once this is be green (I'd hope so), I can merge, which should close #37670

**Pull request merged**